analysis/study_definition.py

```diff
@@ -1,33 +1,191 @@
-from cohortextractor import StudyDefinition, patients, codelist, codelist_from_csv
+from cohortextractor import StudyDefinition, patients, codelist, codelist_from_csv, filter_codes_by_category
 
-chronic_cardiac_disease_codes = codelist_from_csv(
-    "codelists/opensafely-chronic-cardiac-disease.csv", system="ctv3", column="CTV3ID"
-)
-chronic_liver_disease_codes = codelist_from_csv(
-    "codelists/opensafely-chronic-liver-disease.csv", system="ctv3", column="CTV3ID"
-)
-salbutamol_codes = codelist_from_csv(
-    "codelists/opensafely-asthma-inhaler-salbutamol-medication.csv",
-    system="snomed",
-    column="id",
-)
-systolic_blood_pressure_codes = codelist(["2469."], system="ctv3")
-diastolic_blood_pressure_codes = codelist(["246A."], system="ctv3")
+from codelists import *
+
+def first_diagnosis_in_period(dx_codelist):
+    return patients.with_these_clinical_events(
+        dx_codelist,
+        returning="date",
+        find_first_match_in_period=True,
+        include_month=True,
```

Comment:
why do we need dates? Is binary yes/no enough? Checking from a data minimisation point of view and I can't see anything in protocol that requires it

Reply:
We compare dates of diagnosis to determine how to label people where diagnosis has changed.